### ISSUE TITLE
storage: Ranges accumulate in local dev node

### ISSUE THREAD

**Issue by bdarnell** [cockroachdb/cockroach#5984]
In the course of developing cockroachdb/cockroach-python#15, I ran the SQLAlchemy test suite against a persistent local dev node (the tests expect a database to already be running at a static address and so they're not very amenable to the ephemeral testserver approach we use in `cockroach-go`). This creates and destroys a lot of tables, so my local server now has 4974 ranges and approximately no data. 

It would be nice to do something about these ranges in advance of full merge support. A simple solution is to disable splits (or make the max range size huge) for single-node clusters (another use case for an explicit indicator that a cluster is in single-node mode). 

This is enough of a special case (range is empty and we can be sure that no new writes will come in during the merge) that we might be able to merge these ranges without solving the general merge problem. This would be good for shared clusters that are used for this kind of workload (which I would expect to see at a lot of companies based on experience with mysql shops).

**Comment by spencerkimball** [comment]
This doesn't seem gated on 1.0. @bdarnell I'm removing the milestone.

**Comment by bdarnell** [comment]
We're building proper merge support for 2.1, so I'm going to go ahead and call this a duplicate of #2433.
